src/fdc/DiskImageUtils.hh

```diff
@@ -24,13 +30,31 @@ struct MSXBootSector {
 	Endian::L16                 sectorsFat;    // +22 sectors per FAT
 	Endian::L16                 sectorsTrack;  // +24 sectors per track
 	Endian::L16                 nrSides;       // +26 number of side
-	Endian::L16                 hiddenSectors; // +28 not used
-	std::array<uint8_t, 2>      pad1;          // +30
-	Endian::L32                 nrSectorsBig;  // +32 nb of sectors on this disk (>= 65536)
-	std::array<uint8_t, 2>      pad2;          // +32
-	uint8_t                     extendedBootSignature;  // +38
-	Endian::UA_L32              vol_id;        // +39
-	std::array<uint8_t, 512-43> pad3;          // +43
+	union {
+		struct {
+			Endian::L16                 hiddenSectors; // +28 not used
+			std::array<uint8_t, 9>      pad1;          // +30
+			Endian::UA_L32              vol_id;        // +39
+			std::array<uint8_t, 512-43> pad2;          // +43
+		} dos1;
+		struct {
+			Endian::L16                 hiddenSectors; // +28 not used
+			std::array<uint8_t, 9>      pad1;          // +30
+			Endian::UA_L32              vol_id;        // +39
+			std::array<uint8_t, 512-43> pad2;          // +43
+		} dos2;
```

Comment:
Currently the `dos1` and `dos2` fields of the union are identical. But I assume the intention was to leave out the `vol_id` field from `dos1`?